internal/dataagreement/dataagreements.go

```diff
@@ -283,3 +324,73 @@ func (darepo *DataAgreementRepository) GetAll() ([]DataAgreement, error) {
 	}
 	return results, nil
 }
+
+// GetAllDataAgreementsWithLatestRevisionsObbjectData
+func GetAllDataAgreementsWithLatestRevisionsObbjectData(organisationId string) ([]DataAgreementWithObjectData, error) {
```

Comment:
Why do we need a complicated pipeline for this ?

1. Find all the data agreements that matches default filter
2. Iterate through them, if it is draft, fetch the latest revision if available.
3. Convert object data to data agreement

Note: Pipelines should only be used after discussion and complicated cases.

Reply:
Ok.